mavutil.py

```diff
@@ -898,15 +902,25 @@ def recv(self,n=None):
             if e.errno in [ errno.EAGAIN, errno.EWOULDBLOCK, errno.ECONNREFUSED ]:
                 return ""
             raise
-        if self.udp_server or self.broadcast:
+        if self.udp_server:
+            self.clients.add(new_addr)
+            self.clients_last_alive[new_addr] = time.time()
+        elif self.broadcast:
             self.last_address = new_addr
         return data
 
     def write(self, buf):
         try:
             if self.udp_server:
-                if self.last_address:
-                    self.port.sendto(buf, self.last_address)
+                current_time = time.time()
+                to_remove = set()
+                for address in self.clients:
+                    if current_time - self.clients_last_alive[address] <= self.timeout:
```

Comment:
it's a better habit to add times instead of subtracting them to deal with overflows.

if last alive + self.timeout < current_time...

Reply:
(probably won't matter in python, but the habit is the point)